exercises/poker/test/Tests.hs

```diff
@@ -0,0 +1,170 @@
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Data.List         (sort)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import Poker (bestHands)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "bestHands" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion = sort <$> (bestHands input) `shouldBe` sort <$> expected
+
+data Case = Case { description :: String
+                 , input       :: [String]
+                 , expected    :: Maybe [String]
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "single hand always wins"
+               , input       = ["4S 5S 7H 8D JC"]
+               , expected    = Just ["4S 5S 7H 8D JC"]
+               }
+        , Case { description = "highest card out of all hands win"
+               , input       = [ "4D 5S 6S 8D 3C"
+                               , "2S 4C 7S 9H 10H"
+                               , "3S 4S 5D 6H JH"]
+               , expected    = Just ["3S 4S 5D 6H JH"]
+               }
+        , Case { description = "a tie has multiple winners"
+               , input       = [ "4D 5S 6S 8D 3C"
+                               , "2S 4C 7S 9H 10H"
+                               , "3S 4S 5D 6H JH"
+                               , "3H 4H 5C 6C JD"]
+               , expected    = Just [ "3S 4S 5D 6H JH"
+                                    , "3H 4H 5C 6C JD"]
+               }
+        , Case { description =  "multiple hands with the same high cards, tie compares next highest ranked, down to last card"
+               , input       = [ "3S 5H 6S 8D 7H"
+                               , "2S 5D 6D 8C 7S"]
+               , expected    = Just ["3S 5H 6S 8D 7H"]
+               }          
+        , Case { description = "one pair beats high car"
```

Comment:
not high car

Reply:
should be this "multiple hands with the same high cards, tie compares next highest ranked, down to last card" right?

Reply:
no. it should be "one pair beats high card"

Reply:
Ohh, apparently I'm blind